### PR TITLE
chore(index): add catch handle

### PR DESCRIPTION
**Pull request by zacksleo**

### CODE REVIEW

**Review by cheeaun — APPROVED** [review]

**Review by cheeaun — COMMENTED** [review]
I realise that `new` is actually not needed 😅 Would be great if this PR includes changes for all other instances as well 😄

**Comment by zacksleo** [comment]
OK, I will fix it later.

**Comment by zacksleo** [comment]
Updated. Jest test with travis-ci  were added.

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
+// For a detailed explanation regarding each configuration property, visit:
+// https://jestjs.io/docs/en/configuration.html
+
+module.exports = {
```
Comment (cheeaun) [inline jest.config.js]: I personally prefer Jest config to be inside `package.json`, and also all these comments are not needed :) Thanks a lot!

**Review by cheeaun — CHANGES_REQUESTED** [review]

File: jest.config.js (hunk 1 above)
Comment (zacksleo) [reply jest.config.js]: This may be because of the new version of jest. Test not work well when I install jest using the config in the package.json.

Comment (zacksleo) [reply jest.config.js]: I'll try later.

**Review by cheeaun — CHANGES_REQUESTED** [review]
Looks like there are two warnings, from jest in travis:

> Option "scriptPreprocessor" was replaced by "transform", which support multiple preprocessors.

> Unknown option "testFileExtensions" with value ["js"] was found.
> This is probably a typing mistake. Fixing it will remove this message.

**Comment by zacksleo** [comment]
Done

**Review by cheeaun — APPROVED** [review]

**Pull request merged**